src/dynamics/mpas/dyn_comp.F90

```diff
@@ -776,21 +853,70 @@ subroutine set_default_constituent()
 
         call mpas_dynamical_core % get_variable_pointer(scalars, 'state', 'scalars', time_level=1)
 
-        ! `i` is indexing into `scalars`, so it is regarded as MPAS scalar index.
-        do i = 1, num_advected
-            ! Vertical index order is reversed between CAM-SIMA and MPAS.
-            do k = 1, pver
-                scalars(i, k, 1:ncells_solve) = &
-                    constituents(:, pver - k + 1, mpas_dynamical_core % map_constituent_index(i))
+        if (trim(adjustl(direction)) == 'e' .or. trim(adjustl(direction)) == 'export') then
+            do i = 1, ncells_solve
+                if (conversion .and. any(is_conversion_needed)) then
+                    ! The summation term of equation 8 in doi:10.1029/2017MS001257.
+                    ! Using equation 7 here is not possible because it requires all constituent mixing ratio to be moist
+                    ! on the RHS of it. There is no such guarantee in CAM-SIMA.
+                    sigma_all_q(:) = phys_state % pdel(i, :) / phys_state % pdeldry(i, :)
```

Comment:
Great suggestion. This would likely save us many CPU cycles as well.

Addressed in https://github.com/ESCOMP/CAM-SIMA/pull/304/commits/ee5cedb18d3f9107fb894b2976640047875b23b1.